test/controllers/hosts_controller_test.rb

```diff
@@ -1263,38 +1263,49 @@ class Host::Valid < Host::Managed; end
     assert_template 'review_before_build'
   end
 
-  test 'template_used returns templates with interfaces' do
-    @host.setBuild
-    nic=FactoryGirl.create(:nic_managed, :host => @host)
-    attrs = host_attributes(@host)
-    attrs[:interfaces_attributes] = nic.attributes.except 'updated_at', 'created_at', 'attrs'
-    ActiveRecord::Base.any_instance.expects(:destroy).never
-    ActiveRecord::Base.any_instance.expects(:save).never
-    xhr :put, :template_used, {:provisioning => 'build', :host => attrs, :id => @host.id }, set_session_user
-    assert_response :success
-    assert_template :partial => '_provisioning'
-  end
+  describe '#template_used' do
+    setup do
+      @host.setBuild
+      ActiveRecord::Base.any_instance.expects(:destroy).never
```

Comment:
dont you need to remove this mocking once the tests are done? this might remain after the test is done?

Reply:
Doesn't look like that's the case, other tests are able to `.destroy` just fine after this, the expectations are automatically cleared after each of the test.